aiidalab/app.py

```diff
@@ -90,6 +92,17 @@ class AiidaLabApp(traitlets.HasTraits):
     busy = traitlets.Bool(readonly=True)
     modified = traitlets.Bool(readonly=True, allow_none=True)
 
+    @dataclass
+    class AppRegistryData:
```

Comment:
It seems you're fond of the "have a 'package' (here the class `AiidaLabApp`) come with everything inside it may need" :) What I'm trying to say - is there a specific reason you're not putting these "internal" classes outside `AiidaLabApp`? Having a bit more modularity ...

Reply:
The classes are tightly linked to the `AiidaLabApp` class and are not supposed to be used outside of that context, hence why I've made them inner classes. This also reduces the "API surface" overall. However, I think this is mostly a matter of taste and we could take a different approach, but I would prefer to just leave it as is unless you strongly object.

Reply:
No strong objections. It's merely a combination of a sudden realization and stating it "out loud" to start a small talk :)

That said, I completely agree with your assessment here and reasoning behind it 👍